packages/platform-android/src/commands/runAndroid/runOnEmulator.ts

```diff
@@ -26,29 +26,62 @@ function toPascalCase(value: string) {
 
 type AndroidProject = NonNullable<Config['project']['android']>;
 
-async function runOnAllDevices(
+function logStartEmulatorError(error: Error) {
+  if (error.message) {
+    logger.error(
+      `Failed to launch emulator. Reason: ${chalk.dim(error.message)}`,
+    );
+  } else {
+    logger.error('Failed to launch emulator.');
+  }
+
+  logger.warn(
+    'Please launch an emulator manually or connect a device. Otherwise the app may fail to launch.',
+  );
+}
+
+async function startEmulator(adbPath: string, deviceName?: string) {
+  if (deviceName) {
+    logger.info('Launching emulator...');
+
+    try {
+      const deviceListAfterEmulatorLaunched = await launchEmulator(
+        deviceName,
+        adbPath,
+      );
+
+      return deviceListAfterEmulatorLaunched;
+    } catch (error) {
+      return logStartEmulatorError(error);
+    }
+  }
+
+  const devices = adb.getDevices(adbPath);
+  if (devices.length > 0) {
+    return devices;
+  }
+
+  logger.info('Launching emulator...');
+
+  try {
+    const deviceListAfterAnyEmulatorLaunched = await launchAnyEmulator(adbPath);
+
+    return deviceListAfterAnyEmulatorLaunched;
+  } catch (error) {
+    console.log({error});
+    return logStartEmulatorError(error);
```

Comment:
Do we need a separate `logStartEmulatorError` function here? How about we just handle it inline?

Reply:
I see, the reason it was done was due to line 55, where it is being referenced as well.

I would wrap `launchAnyEmulator` and `launchEmulator` in the same `try/catch` block to address this.

Reply:
To be honest, it could've been simplified to:
```js
async function startEmulator(adbPath: string, deviceName?: string) {
    try {
        if (deviceName) {
            logger.info('Launching ${deviceName} emulator...');
            return launchEmulator(
                deviceName,
                adbPath,
            );
        } else {
            logger.info('Launching first available emulator...');
            return launchAnyEmulator(adbPath);
        }
    } catch (error) {
        if (error.message) {
            logger.error(
                `Failed to launch emulator. Reason: ${chalk.dim(error.message)}`,
            );
        } else {
            logger.error('Failed to launch emulator.');
        }

        logger.warn(
            'Please launch an emulator manually or connect a device. Otherwise the app may fail to launch.',
        );
    }
}
```
and in such case, I am not even sure if we need this function - maybe we can place it directly inside `runAndroid` instead.